ansible_base/rbac/api/permissions.py

```diff
@@ -0,0 +1,104 @@
+import logging
+
+from django.db.models import Model
+from django.http import Http404
+from rest_framework.permissions import SAFE_METHODS, BasePermission, DjangoObjectPermissions
+
+from ansible_base.lib.utils.models import is_add_perm
+from ansible_base.rbac import permission_registry
+from ansible_base.rbac.evaluations import has_super_permission
+
+logger = logging.getLogger('ansible_base.rbac.api.permissions')
+
+
+class IsSystemAdminOrAuditor(BasePermission):
+    """
+    Allows write access only to system admin users.
+    Allows read access only to system auditor users.
+    """
+
+    def has_permission(self, request, view):
+        if not (request.user and request.user.is_authenticated):
+            return False
+        if request.method in SAFE_METHODS:
+            return has_super_permission(request.user, 'view')
+        return has_super_permission(request.user)
+
+
+class AuthenticatedReadAdminChange(IsSystemAdminOrAuditor):
+    "Any authenticated user can view, but only admin users can do CRUD"
+
+    def has_permission(self, request, view):
+        if not (request.user and request.user.is_authenticated):
+            return False
+        if request.method in SAFE_METHODS:
+            return True
+        return has_super_permission(request.user)
+
+
+class AnsibleBaseObjectPermissions(DjangoObjectPermissions):
```

Comment:
Those (and the AWX permission class) inherit from `BasePermission` which is a super class of `DjangoObjectPermissions`. Yes, I am using some methods from `DjangoObjectPermissions`. This is very dear to me, because it has methods to translate HTTP method & model to permissions. As hard as it was to work that in, I am still clinging for dear life to have something as close to standard as I can. That means using the model `_meta` permissions, and accepting the some call patterns from that.

Now that you mentioned it, I filed https://github.com/ansible/django-ansible-base/issues/221 to make it look _more_ like the DRF class. We need to write this on the whiteboard like Bart Simpson in detention "Don't make things custom when there is no need to"